### PR TITLE
ICMSLST-2135 - Gov Notify

### PR DESCRIPTION
**Pull request by marcuspp**
Initial implementation of Gov Notify
 - Gov notify backend added
 - Access requested email updated to use gov notify
 - Backend not set by default to GOV Notify this will be done once Gov Notify is 'Live'
 - Support added to override recipients of emails on dev/local to aid with manual testing
 - Logging handled by `django-celery-results`
 - Removed `django-ses`

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 boto3==1.21.29
-celery[redis]==5.2.7
+celery[redis]==5.3.1
```
Comment (marcuspp) [inline requirements-base.txt:2]: Came across this issue during testing https://github.com/celery/billiard/issues/377 Upgrading celery fixed the issue

File: requirements-base.txt [hunk 2]
```diff
@@ -26,6 +25,7 @@ notifications-python-client==8.0.1
 openpyxl==3.0.7
 oracledb==1.2.0
 phonenumbers==8.12.12
+pytz==2023.3
```
Comment (marcuspp) [inline requirements-base.txt:28]: Was a dependancy of celery 5.2.7 and not 5.3.1. Is used by `add_commodity_data` script

File: web/mail/backends.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+from web.mail.api import send_email
+
+logger = logging.getLogger(__name__)
+
+
+class GovNotifyEmailBackend(BaseEmailBackend):
+    def send_messages(self, email_messages: list) -> None:
+        for message in email_messages:
+            for recipient in message.recipients():
+                logger.info(f"Sending {message.name.label} email to {recipient}")
```
Comment (MattHolmes123) [inline web/mail/backends.py:15]: Logging should us the old %-style formatting.
It's for backward compatability mainly, but the formatting doesn't take place when logging is disabled if you do it the %-style way.
(I know a few f-string formatting lines have snuck in to ICMS).

File: web/mail/backends.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import structlog as logging
+from django.core.mail.backends.base import BaseEmailBackend
+
+from web.mail.api import send_email
+
+logger = logging.getLogger(__name__)
+
+
+class GovNotifyEmailBackend(BaseEmailBackend):
+    def send_messages(self, email_messages: list) -> None:
```
Comment (MattHolmes123) [inline web/mail/backends.py:12]:
```
from web.mail.messages import GOVNotifyEmailMessage
def send_messages(self, email_messages: list[GOVNotifyEmailMessage]) -> None:
```

File: web/mail/messages.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        } | self.get_context()
+
+    def get_template_id(self) -> UUID:
+        return EmailTemplate.objects.get(name=self.name).gov_notify_template_id
+
+
+class AccessRequestEmail(GOVNotifyEmailMessage):
+    name = EmailTypes.ACCESS_REQUEST
+
+    def __init__(self, access_request, *args, **kwargs):
```
Comment (MattHolmes123) [inline web/mail/messages.py:43]: Any reason you've moved away from doing it as a named kwarg?

e.g.
```python
def __init__(self, *args, access_request: AccessRequest, **kwargs):
```

File: web/mail/messages.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+from django.conf import settings
+from django.core.mail import EmailMessage, SafeMIMEMultipart
+
+from .constants import EmailTypes
+from .models import EmailTemplate
+
+
+class GOVNotifyEmailMessage(EmailMessage):
+    name = None | EmailTypes
```
Comment (MattHolmes123) [inline web/mail/messages.py:11]:
```python
from typing import ClassVar
name: ClassVar[EmailTypes | None] = None
```
In your version name get assigned a value of `<class 'types.UnionType'>`

File: web/mail/messages.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        return {
+            "icms_url": settings.DEFAULT_DOMAIN,
+            "icms_contact_email": settings.ILB_CONTACT_EMAIL,
+            "icms_contact_phone": settings.ILB_CONTACT_PHONE,
+            "subject": self.subject,
+            "body": self.body,
+        } | self.get_context()
+
+    def get_template_id(self) -> UUID:
+        return EmailTemplate.objects.get(name=self.name).gov_notify_template_id
```
Comment (MattHolmes123) [inline web/mail/messages.py:37]: Don't think it will be a problem but this will hit the db everytime any derived class is initialised.

**Review by MattHolmes123 — APPROVED** [review]

**Pull request merged**